wordpress-mobile/WordPress-Android
# Restore user agent for WebViewActivity

Fixes the _Site → Blog Posts_ part of #9625. 

## Findings

The changes in Automattic/wp-calypso#31160 expected the Android app to have `"wp-android"` in its user agent. You can see this being done in the Automattic/wp-calypso#31830 PR. See also related discussion in p4TIVU-9dz-p2.

It looks like for `WebViewActivity`, we never change the user agent at all. We removed that in #1312. At that time, the user agent fully replaced the default. It looked like this:

```
wp-android/4.7 (Android 6.0; en_US; <...>)
```

After the changes in #3410, the user agent is now the default plus the custom `"wp-android"` text. It looks like this:

```
Mozilla/5.0 (Linux; Android 9; Pixel 3 Build/PQ2A.190305.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/73.0.3683.90 Mobile Safari/537.36 wp-android/<version>
```

## Solution 

I believe it is now safe to restore the user agent and that's what was done here. This appends `"wp-android/<version>"` to the default user agent. 

```
// Before
Mozilla/5.0 (Linux; Android 9; Pixel 3 Build/PQ2A.190305.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/73.0.3683.90 Mobile Safari/537.36

// After
Mozilla/5.0 (Linux; Android 9; Pixel 3 Build/PQ2A.190305.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/73.0.3683.90 Mobile Safari/537.36 wp-android/<version>
```

| Before | After |
|--------|-------|
|   ![2019-04-15 14 08 45](https://user-images.githubusercontent.com/198826/56163973-f27f4d00-5f8c-11e9-8ffe-5ab2a8b8f22a.png) |    ![2019-04-15 14 41 31](https://user-images.githubusercontent.com/198826/56163986-f7440100-5f8c-11e9-90d9-73277002f76a.png) |

## Testing

1. Go to a WPCom site's _Blog Posts_
2. Click on a post's _View_ button. 

Validate that the Calypso UI is no longer shown. 

Please also test for possible regressions that you can remember that's related to the removal of the user agent. 

## Release notes
